dashboard-ui/src/pages/home.tsx

```diff
@@ -326,38 +401,33 @@ type DisplayItemsProps = {
   workload: Workload;
   namespace: string;
   fetching: boolean;
-  items: {
-    id: string;
-    metadata: {
-      uid: string;
-      namespace: string;
-      name: string;
-      creationTimestamp: any;
-      deletionTimestamp?: Date;
-    };
-  }[] | undefined | null;
+  items:
+  | WorkloadItem[]
+  | undefined
+  | null;
   ownershipMap: Map<string, string[]>;
 };
 
 const DisplayItems = ({
-  workload, namespace, fetching, items, ownershipMap,
+  workload,
+  namespace,
+  fetching,
+  items,
+  ownershipMap,
 }: DisplayItemsProps) => {
-  const { kubeContext, workloadFilter } = useContext(Context);
-
-  // filter items
-  const filteredItems = items?.filter((item) => {
-    // remove deleted items
-    if (item.metadata.deletionTimestamp) return false;
+  const { kubeContext, search, workloadFilter } = useContext(Context);
 
-    // remove items not in filtered namespace
-    return (namespace === '' || item.metadata.namespace === namespace);
-  });
+  // destructuring items array due to read-only array issue
+  const filteredItems = items ? [...items] : [];
 
-  const ids = filteredItems?.map((item) => item.metadata.uid) || [];
+  const ids = items?.map((item) => item.metadata.uid) || [];
```

Comment:
Why do you need the list of unfiltered item ids instead of filtered item ids here?